src/grpc/config/hpr_cs_skf_stream_worker.erl

```diff
@@ -70,65 +68,42 @@ handle_cast(Msg, State) ->
 
 handle_info(?INIT_STREAM, #state{conn_backoff = Backoff0} = State) ->
     lager:info("connecting"),
-    case hpr_cs_conn_worker:get_connection() of
-        undefined ->
+    {PubKey, SigFun} = persistent_term:get(?HPR_KEY),
+    PubKeyBin = libp2p_crypto:pubkey_to_bin(PubKey),
+    SKFStreamReq = hpr_skf_stream_req:new(PubKeyBin),
+    SignedSKFStreamReq = hpr_skf_stream_req:sign(SKFStreamReq, SigFun),
+    SignedSKFStreamReqMap = hpr_skf_stream_req:to_map(SignedSKFStreamReq),
+    StreamOptions = #{channel => config_channel},
+
+    case helium_config_session_key_filter_client:stream(SignedSKFStreamReqMap, StreamOptions) of
+        {ok, Stream} ->
+            lager:info("stream initialized"),
+            {_, Backoff1} = backoff:succeed(Backoff0),
+            {noreply, State#state{stream = Stream, conn_backoff = Backoff1}};
+        {error, _E} ->
             {Delay, Backoff1} = backoff:fail(Backoff0),
-            lager:error("failed to get connection sleeping ~wms", [Delay]),
+            lager:error("failed to get stream sleeping ~wms", [Delay]),
             _ = erlang:send_after(Delay, self(), ?INIT_STREAM),
-            {noreply, State#state{conn_backoff = Backoff1}};
-        Connection ->
-            {_, Backoff1} = backoff:succeed(Backoff0),
-            #{http_connection := Pid} = Connection,
-            _Ref = erlang:monitor(process, Pid, [{tag, {'DOWN', ?MODULE}}]),
-            lager:info("connected"),
-            {ok, Stream} = grpc_client:new_stream(
-                Connection, 'helium.config.session_key_filter', stream, client_config_pb
-            ),
-            %% Sending SKF Stream Request
-            {PubKey, SigFun} = persistent_term:get(?HPR_KEY),
-            PubKeyBin = libp2p_crypto:pubkey_to_bin(PubKey),
-            SKFStreamReq = hpr_skf_stream_req:new(PubKeyBin),
-            SignedSKFStreamReq = hpr_skf_stream_req:sign(SKFStreamReq, SigFun),
-            ok = grpc_client:send_last(
-                Stream, hpr_skf_stream_req:to_map(SignedSKFStreamReq)
-            ),
-            lager:info("stream initialized"),
-            self() ! ?RCV_CFG_UPDATE,
-            {noreply, State#state{stream = Stream, conn_backoff = Backoff1}}
+            {noreply, State#state{conn_backoff = Backoff1}}
     end;
+%% GRPC stream callbacks
+handle_info({data, _StreamID, SKFStreamRes}, State) ->
+    lager:debug("sfk update"),
+    ok = process_res(hpr_skf_stream_res:from_map(SKFStreamRes)),
+    {noreply, State};
 handle_info(
-    ?RCV_CFG_UPDATE,
-    #state{
-        stream = Stream, conn_backoff = Backoff0
-    } = State
+    {'DOWN', Ref, process, Pid, _Reason},
```

Comment:
I don't see any monitor above, does helium_config_session_key_filter_client:stream monitors by itself?

Reply:
https://github.com/novalabsxyz/grpcbox/blob/master/src/grpcbox_client_stream.erl#L50

The monitor is added when the stream is created.